indexclient/client.pyi

```diff
@@ -0,0 +1,207 @@
+from typing_extensions import Protocol, TypedDict
+
+from typing import (
+    Dict,
+    Iterable,
+    List,
+    Tuple,
+    Optional,
+    Union,
+    Any,
+    NamedTuple,
+    Literal,
+)
+
+from requests import Response
+
+IndexFileLocationType = Literal[
+    "aws_archive",
+    "aws_s3",
+    "aws_s3_gl",
+    "aws_s3_rr",
+    "blackpearl-primary",
+    "cleversafe",
+    "primary-backup",
+]
+
+IndexFileState = Literal[
+    "backing_up",
+    "backuped",
+    "deleted",
+    "error",
+    "failed",
+    "registered",
+    "uploaded",
+    "uploading",
+    "validated",
+    "validating",
+]
+
+class URLMetadataTuple(NamedTuple):
+    url: str
+    type: IndexFileLocationType
+    state: IndexFileState
+
+class UrlMetadataDict(TypedDict):
+    state: IndexFileState
+    type: IndexFileLocationType
```

Comment:
Does this allow for other dictionary keys? For example we use `md5` as a key for backed up urls in the runners.

Reply:
yea, typeddicts are essentially dictionaries with known keys